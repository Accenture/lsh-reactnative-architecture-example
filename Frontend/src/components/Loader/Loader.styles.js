const loaderStyles = {
  flex: 1,
  alignItems: 'center',
  justifyContent: 'center',
};
export default loaderStyles;
