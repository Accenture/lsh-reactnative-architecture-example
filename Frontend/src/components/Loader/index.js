import React from 'react';
import PropTypes from 'prop-types';
import { View, ActivityIndicator } from 'react-native';
import styles from './Loader.styles';

const Loader = ({ size }) => (
  <View style={styles}>
    <ActivityIndicator size={size} />
  </View>
);

Loader.defaultProps = {
  size: 'large',
};

Loader.propTypes = {
  size: PropTypes.string,
};

export default Loader;
