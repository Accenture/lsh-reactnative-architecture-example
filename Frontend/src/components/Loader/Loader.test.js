import React from 'react';
import { shallow } from 'enzyme';
import Loader from './index';

describe('Tests loader component', () => {
  it('matches snapshot', () => {
    const comp = shallow(<Loader />);
    expect(comp).toMatchSnapshot();
  });
});
