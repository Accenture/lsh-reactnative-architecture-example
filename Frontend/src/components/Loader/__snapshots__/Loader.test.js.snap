// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests loader component matches snapshot 1`] = `
<Component
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <ActivityIndicator
    animating={true}
    color="#999999"
    hidesWhenStopped={true}
    size="large"
  />
</Component>
`;
